File: static/js/jumio.js
$(function(){
  console.log("jumio loaded");
  var marchant_token="fcf1eec3-728d-4f8a-8811-5b8e0e534597";
  var api_secret = "9mnQyVj1ppiyVESYroDHZS23Z9OfQ9GS";
  var success_url="https://simtransfer.com/jumiopass/";
  var error_url="https://simtransfer.com/jumiofail/";

  var post_data={
    "successurl":success_url,
    "error_url":error_url
  };

  var user_data={
    marchant_token:api_secret
  };

  var data_hash=window.btoa(JSON.stringify(post_data));

  var user_hash=window.btoa(JSON.stringify(user_data));

  console.log("Base64 Data: "+data_hash);
  console.log("Base64 User: "+user_hash);

  //jumio_request(data_hash,user_hash);
  cors_test(data_hash,user_hash,success_url,error_url);


});


function base_auth(item,value){
  var token = item+":"+value;
  var hash = //window.btoa(unescape(encodeURIComponent(token)));
  window.btoa(token);
  console.log("Hash: "+hash);
  return hash;
}

function jumio_request(data_hash,user_hash){
  var jumio_url="https://netverify.com/api/netverify/v2/initiateNetverify";

  $.ajax({
    url:jumio_url,
    type:"POST",
    contentType:"application/json",
    crossDomain: true,
    data:{
      "Authorization ":"Basic "+data_hash
    },

    headers:{
       "Access-Control-Allow-Origin":"https://netverify.com/api/netverify/v2/initiateNetverify",
       Authorization:"Basic "+user_hash,
       Accept: "application/json",
       //"Content-Type":"application/json",
       "User-Agent":"MyCompany MyApp/1.0.0",
    },

    //beforeSend:function(xhr){
      //xhr.setRequestHeader("Authorization", "Basic "+user_hash);
      //xhr.setRequestHeader("Accept","application/json");
      //xhr.setRequestHeader("Content-Type", "application/json");
      //xhr.setRequestHeader("Accept","application/json");
      //xhr.setRequestHeader("User-Agent","YOURCOMPANYNAME YOURAPPLICATIONNAME/x.x.x");
    //},
    success:function(data){
      console.log("Success: "+data);
    },
    error:function(jqXhr, textStatus, errorThrown){
      console.log("Errors: "+JSON.stringify(jqXhr)+" "+textStatus.toString()+" "+errorThrown.toString());
    }

  });

}


/*


curl -i --user validuser:70e12a10-83c7-11e0-9d78-0800200c9a65 -H "Content-Type: application/json" -H "Accept: application/json" -X POST -d '{"person":{"name":"bob"}}' http://mysite.com/api.php

curl -H




curl --user fcf1eec3-728d-4f8a-8811-5b8e0e534597:password 9mnQyVj1ppiyVESYroDHZS23Z9OfQ9GS  https://netverify.com/api/netverify/v2/initiateNetverify

*/


/*
Data:
eyJzdWNjZXNzdXJsIjoiaHR0cHM6Ly9zaW10cmFuc2Zlci5jb20vanVtaW9wYXNzLyIsImVycm9yX3VybCI6Imh0dHBzOi8vc2ltdHJhbnNmZXIuY29tL2p1bWlvZmFpbC8ifQ==

User:
eyJtYXJjaGFudF90b2tlbiI6IjltblF5VmoxcHBpeVZFU1lyb0RIWlMyM1o5T2ZROUdTIn0=

*/

File: remit/jumio.py
import requests
import json
from remit import settings as settings
from remit import utils as utils

rest_url = settings.JUMIO_URL
token = settings.JUMIO_TOKEN
secret_key = settings.JUMIO_SECRET
success_url=settings.JUMIO_SUCCESS_URL
error_url=settings.JUMIO_ERROR_URL
user_agent=settings.JUMIO_USER_AGENT
callback_url=settings.JUMIO_CALLBACK


class Jumio:
    """
    Handle jumio data
    """

    def auth_token(self):
        """
        get and return auth token
        """
        response_data={}
        scan_ref = utils.random_string_generator(15)

        headers = {
            'accept':'application/json',
            'content-Type':'application/json',
            'user-agent':user_agent,
            'Content-Length':'0'
        }

        user_data = {
            'merchantIdScanReference':scan_ref,
            'successurl':success_url,
            'error_url':error_url,
            'callbackUrl':callback_url,
        }

        data = requests.post(
            rest_url,
            headers=headers,
            data=json.dumps(user_data),
            auth=(token,secret_key)
        )

        #
        try:
            data = requests.post(
                rest_url,
                headers=headers,
                data=json.dumps(user_data),
                auth=(token,secret_key)
            )
            print 'jumio success: ',str(data.json())
        except Exception as e:
            print 'Jumio Error: ',str(e)
        #

        response = data.json()

        response_data['scan_ref']=scan_ref
        response_data['auth_token']=response[u'authorizationToken']

        #print str(response)

        #return response[u'authorizationToken']
        return response_data
